in_splunk: Add descriptions for recently enhancement of in_splunk (#1387)
fluent/fluent-bit-docs: pipeline/inputs/splunk.md

| buffer_max_size          | Specify the maximum buffer size in KB to receive a JSON message.                                                                              | 4M      |
| buffer_chunk_size        | This sets the chunk size for incoming incoming JSON messages. These chunks are then stored/managed in the space available by buffer_max_size. | 512K    |
| successful_response_code | It allows to set successful response code. `200`, `201` and `204` are supported.                                                              | 201     |

> | splunk\_token           | Add an Splunk token for HTTP HEC. If specified multiple tokens with commas, it will be divided into each of tokens and they will be used for authentication of HTTP HEC. |         |

Reviewer: Is the division across tokens equal or is it just use the first one until it is maxed, then the second, and so on?
```suggestion
| splunk\_token           | Specify a Splunk token for HTTP HEC authentication. If multiple tokens are specified (with commas and no spaces), usage will be divided across each of the tokens. |         |
```

Author: With the current implementation, the division is across tokens equal.
Splunk token is able to specify only one at sender(HEC client). So, the specified tokens are traversed until the max or matched.

> | store\_token\_in\_metadata | Store Splunk HEC tokens as a part of metadata. If set as false, they will be stored as a part of records.                                   | true    |

Reviewer: ```suggestion
| store\_token\_in\_metadata | Store Splunk HEC tokens in the Fluent Bit metadata. If set false, they will be stored as normal key-value pairs in the record data.                                   | true    |
```

> | store\_token\_to\_metadata | Store Splunk HEC tokens to matadata. If set as false, they will be stored into records.                                                     | true    |

Reviewer: ```suggestion
| store\_token\_to\_metadata | Store Splunk HEC tokens as metadata. If set as false, they will be stored as records.                                                     | true    |
```

Reviewer: Not sure what you mean here by metadata - is that on the Splunk side so nothing to do with FB metadata/records?

Author: It's a fluent-bit side of metadata:

```
[[[timestamp, {metadata}], {record objects}]]
```

ref: https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1.5#forward-mode

> | splunk\_token\_key       | Use the specified key for storing the Splunk token for HTTP HEC. This is effective when `store_token_in_metadata` as false.                   | @splunk_token |

Reviewer: ```suggestion
| splunk\_token\_key       | Use the specified key for storing the Splunk token for HTTP HEC. This is only effective when `store_token_in_metadata` is false.                   | @splunk_token |
```

> | splunk\_token\_key       | Add a key for storing Splunk token for HTTP HEC. This is effective when `store_token_to_metadata` as false.                                   | @splunk_token |

Reviewer: Is this the key you use for the token?
```suggestion
| splunk\_token\_key       | Use the specified key for storing the Splunk token for HTTP HEC. This is effective when `store_token_to_metadata` as false.                                   | @splunk_token |
```

Author: Yes, it is. It will be used to store ingested Splunk HEC tokens for authentications.
